pandas/io/formats/csvs.py

```diff
@@ -50,7 +50,17 @@ def __init__(self, obj, path_or_buf=None, sep=",", na_rep='',
 
         self.header = header
         self.index = index
-        self.index_label = index_label
+        # if index label is not explicitly called, index label is True if
+        # header or index is not False; otherwise, index label is set to False
+        if index_label is None:
+            if self.header is False or self.header is None or not self.index:
```

Comment:
```suggestion
            if not (self.header or self.index):
```

Reply:
yeah... thanks @WillAyd ... probably i overthink it... previously i was considering an empty list...

Reply:
So `not self.header or not self.index` then?

Reply:
can you update this

``
self.index_label = self.header or self.index
``

seems equivalent

Reply:
it does seem equivalent though,
```
        header : bool or list of str, default True
            Write out the column names. If a list of strings is given it is
            assumed to be aliases for the column names.
            .. versionchanged:: 0.24.0
```
for instance, since from version 0.24.0, it looks like `header` could be a list, and if a list is given, we don't want to assign this list to `index_label`

Reply:
try ``bool(self.header) or self.index`` then